lib/util/propTypes.js

```diff
@@ -281,6 +283,49 @@ module.exports = function propTypesInstructions(context, components, utils) {
     return ignorePropsValidation;
   }
 
+  function declarePropTypesForTSTypeAnnotation(propTypes, declaredPropTypes) {
+    let foundDeclaredPropertiesList = [];
+    if (propTypes.typeAnnotation.type === 'TSTypeReference') {
+      const typeName = propTypes.typeAnnotation.typeName.name;
+      if (!typeName) {
+        return true;
+      }
+      // the game here is to find the type declaration in the code
+      const candidateTypes = context.getSourceCode().ast.body.filter((item) => item.type === 'VariableDeclaration' && item.kind === 'type');
+      const declarations = flatMap(candidateTypes, (type) => type.declarations);
+
+      // we tried to find either an interface or a type with the TypeReference name
+      const typeDeclaration = declarations.find((dec) => dec.id.name === typeName);
+      const interfaceDeclaration = context.getSourceCode().ast.body.find((item) => item.type === 'TSInterfaceDeclaration' && item.id.name === typeName);
+
+
+      if (typeDeclaration) {
+        foundDeclaredPropertiesList = typeDeclaration.init.members;
+      } else if (interfaceDeclaration) {
+        foundDeclaredPropertiesList = interfaceDeclaration.body.body;
+      } else {
+        // type not found, for example can be an exported type, etc. Can issue a warning in the future.
+        return true;
```

Comment:
it would be great to add test cases to cover this, so we're aware when it changes in the future